iceoryx_examples/callbacks/ice_callbacks_subscriber.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/popo/active_call_set.hpp"
+#include "iceoryx_posh/popo/typed_subscriber.hpp"
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/semaphore.hpp"
+#include "topic_data.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <iostream>
+
+iox::popo::UserTrigger shutdownTrigger;
+iox::posix::Semaphore mainLoopBlocker =
+    iox::posix::Semaphore::create(iox::posix::CreateUnnamedSingleProcessSemaphore, 0).value();
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    shutdownTrigger.trigger();
+}
+
+void shutdownTriggerCallback(iox::popo::UserTrigger* trigger)
+{
+}
```

Comment:
I updated it a bit and added a todo in the issue to rework the example/docu here. The idea of this first example was to provide the reviewers an opportunity to see how this would be used.